### DISCUSSION TITLE
Getting Segmentation fault in all api calls for PBKDF2

### DISCUSSION THREAD

**Discussion by rahulcode60** [openssl/openssl#24710] [Q&A]
I have this code and I am trying to get name of KDF implementation. 
kdf = EVP_KDF_fetch(NULL, "PBKDF2", NULL);
    if (kdf == NULL) {
		fprintf(stderr, "failed to fetch PBKDF\n");
		return 0;
	}

When I call any api on this kdf getting segmentation fault. I tried below apis.
EVP_KDF_get0_name
EVP_KDF_names_do_all
EVP_KDF_gettable_params

**Comment by t8m** [comment]
This is really strange and something that should not happen. You can try running the program in a debugger and look where the segfault happens and what is the backtrace.

**Comment by rahulcode60** [comment]
Getting segmentation fault in api call itself. I tried printing backtrace and it failing at this stage.

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7c86264 in EVP_KDF_get0_name () from /usr/local/lib64/libcrypto.so.3

**Reply by t8m** [reply]
Unfortunately I cannot help you much with this. You'll need to build OpenSSL with debugging information (use -d on Configure command line) to get proper information from the debugger.

**Comment by rahulcode60** [comment]
Okay Thanks. I found the issue was with one of my header files. Closing the discussion.
